app/bundles/CoreBundle/Resources/views/Offline/offline.html.twig

```diff
@@ -0,0 +1,66 @@
+    <meta name="description" content="">
+    <meta name="author" content="">
+    <title>Site is offline</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
+    <link rel="icon" type="image/x-icon" href="{{error.assetBase}}/images/favicon.ico"/>
+    <link rel="stylesheet" href="{{error.assetBase}}/css/libraries.css"/>
+    <link rel="stylesheet" href="{{error.assetBase}}/css/app.css"/>
+</head>
+<body>
+<div class="container">
+    <div class="row">
+        {% if error is defined and error is not empty %}
+        <div class="{{ (error.isPrevious is defined and error.isPrevious is not empty) or (inline is defined and inline is not empty) ? 'col-sm-12' : 'col-sm-offset-2 col-sm-8' }}">
+            <div class="bg-white pa-sm" style=" word-wrap: break-word;{% if (error.isPrevious is not defined or error.isPrevious is empty) and (inline is not defined or inline is empty) %} margin-top:100px;{% endif %}">
+            {% if inline is defined and inline is not empty %}
+                <h3><i class="fa fa-warning fa-fw text-danger pull-left"></i>{{ error.message | raw }}</h3>
+                <h6 class="text-muted">{{ error.file |raw ~ ':'~ error.line | raw }}</h6>
+            {% else %}        
+                <div class="text-center">
+                    <i class="fa fa-warning fa-5x"></i>
+                </div>        
+                <div class="alert alert-danger">
+                    <div>{{ error.message | raw }}</div>
+                    <div class="text-muted small text-right mt-10">
+                        {{ error.file | raw ~ ':'~ error.line | raw }}
+                    </div>
+                </div>
+            {% endif %}
+
+            {% if error.trace is defined and error.trace is not empty  %}
+            <div class="well well-sm" tabindex="-1" style="{{ inline is not defined or inline is empty ? '' : 'max-height: 100px; overflow: scroll;'}}" onclick="this.focus();">
+                {% if error.trace is defined and error.trace is not empty and error.trace is iterable %}
+                    {% include 'traces.html.twig' with {'traces': error.trace, 'root': error.root } %}
+                {% else %}
+                    <pre>{{ error.trace | raw }}</pre>;
+                {% endif %}
+                <div class="clearfix"></div>
+            {% endif %}
+        <div id="previous"></div>
+    </div>
+    {% elseif inline is defined and inline is not empty %}
+    <div class="col-xs-12">
+        <h3><i class="fa fa-warning fa-fw text-danger"></i>{{ message }}</h3>
+        {% if submessage is defined and submessage is not empty %}
+        <h4 class="mt-15">{{ submessage | raw }}</h4>
+        {% endif %}
+    </div>
+    {% else %}
+    <div class="col-sm-offset-3 col-sm-6">
+        <div class="bg-white pa-lg text-center" style="margin-top:100px;">
+            <i class="fa fa-warning fa-5x"></i>
+            <h2>{{ message | raw }}</h2>
```

Comment:
I think we should use the `purify` filter instead of `raw` as I can imagine that some attacker could cause somehow cause an error with some XSS in the message.

Reply:
I tried to use `purify` filter, but I ended up with the message, am I supposed to register this filter somewhere 

![image](https://user-images.githubusercontent.com/12627658/227052438-6652e474-56e5-4c7d-a9bb-41f80777e732.png)

Reply:
Not sure how that's possible. It's used all over the place. Example: https://github.com/mautic/mautic/blob/5.x/app/bundles/CoreBundle/Resources/views/Helper/_detail_row.html.twig#L8

Reply:
I see. The purify filter is added to the Twig engine after this code runs. How about to use https://twig.symfony.com/doc/3.x/filters/escape.html instead of `raw`? It might add even more value then run the HTML and JS in the error page. The developer wants to see the markup and it will be safe.

Reply:
@escopecz `escape` did not do the trick. I looked into other filters  but none seems to be doing what we are looking for

![image](https://user-images.githubusercontent.com/12627658/227198370-74a64d5c-779e-4aca-b021-271fc06f1562.png)